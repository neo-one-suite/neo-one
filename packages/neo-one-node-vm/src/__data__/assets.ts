import { Asset, AssetType, common, crypto, ScriptBuilder } from '@neo-one/client-core';

const ONE_HUNDRED_MILLION = common.fixed8FromDecimal(100000000);

const neoAdmin = crypto.toScriptHash(new ScriptBuilder().emitOp('PUSH1').build());

export const NEO_ASSET_HASH_UINT256 = common.stringToUInt256(common.NEO_ASSET_HASH);

export const NEO_ASSET = new Asset({
  hash: NEO_ASSET_HASH_UINT256,
  type: AssetType.GoverningToken,
  name: '[{"lang":"zh-CN","name":"小蚁股"},{"lang":"en","name":"AntShare"}]',
  amount: ONE_HUNDRED_MILLION,
  precision: 0,
  owner: common.ECPOINT_INFINITY,
  admin: neoAdmin,
  issuer: neoAdmin,
  expiration: 2 * 2000000,
  isFrozen: false,
});

const gasAdmin = crypto.toScriptHash(new ScriptBuilder().emitOp('PUSH0').build());

export const GAS_ASSET_HASH_UINT256 = common.stringToUInt256(common.GAS_ASSET_HASH);

export const GAS_ASSET = new Asset({
  hash: GAS_ASSET_HASH_UINT256,
  type: AssetType.UtilityToken,
  name: '[{"lang":"zh-CN","name":"小蚁币"},{"lang":"en","name":"AntCoin"}]',
  amount: ONE_HUNDRED_MILLION,
  precision: 8,
  owner: common.ECPOINT_INFINITY,
  admin: gasAdmin,
  issuer: gasAdmin,
  expiration: 2 * 2000000,
  isFrozen: false,
});

const assets = {
  [common.NEO_ASSET_HASH]: NEO_ASSET,
  [common.GAS_ASSET_HASH]: GAS_ASSET,
};

// tslint:disable-next-line no-any
export const createGetAsset = (): any =>
  jest.fn(({ hash }) => {
    // tslint:disable-next-line no-any
    const asset = assets[common.uInt256ToString(hash)] as any;
    if (asset === undefined) {
      throw new Error(`Unknown asset ${common.uInt256ToString(hash)}`);
    }

    return asset;
  });
