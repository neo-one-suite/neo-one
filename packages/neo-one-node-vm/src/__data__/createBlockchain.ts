// tslint:disable no-any no-object-mutation readonly-keyword
import { common, Contract, crypto, StorageItem } from '@neo-one/client-core';

type Storages = { [key: string]: StorageItem };

export const createBlockchain = ({ contracts = [] }: { readonly contracts?: ReadonlyArray<Contract> }): any => {
  const blockchain = {
    contract: {},
    storageItem: {},
    settings: {
      vm: {},
    },

    action: {},
    output: {},
    asset: {},
    currentBlock: {
      timestamp: 1518512400,
    },
  } as any;

  if (contracts.length > 0) {
    const scriptHashToContract = contracts.reduce<{ [key: string]: Contract }>((acc, contract) => {
      acc[common.uInt160ToString(crypto.toScriptHash(contract.script))] = contract;
      return acc;
    }, {});
    blockchain.contract.get = jest.fn(async ({ hash }) => {
      const contract = scriptHashToContract[common.uInt160ToString(hash)];
      if (contract === undefined) {
        throw new Error(`Unknown contract: ${common.uInt160ToString(hash)}`);
      }

      return contract;
    });
  }

  const storage: { [key: string]: Storages } = {};
  blockchain.storageItem.add = jest.fn(async (item) => {
    const hash = common.uInt160ToString(item.hash);
    if ((storage[hash] as Storages | undefined) === undefined) {
      storage[hash] = {};
    }

    storage[hash][item.key.toString('hex')] = item;
  });
  blockchain.storageItem.update = jest.fn(async (item, { value }) => {
    storage[common.uInt160ToString(item.hash)][item.key.toString('hex')] = item.update({ value });
  });
  blockchain.storageItem.get = jest.fn(async ({ hash, key }) => {
    const hashStorage = storage[common.uInt160ToString(hash)];
    if ((hashStorage as Storages | undefined) === undefined) {
      throw new Error('Item not found');
    }

    const item = hashStorage[key.toString('hex')] as StorageItem | undefined;
    if (item === undefined) {
      throw new Error('Item not found');
    }

    return item;
  });
  blockchain.storageItem.tryGet = jest.fn(
    async ({ hash, key }) => (storage[common.uInt160ToString(hash)] || {})[key.toString('hex')],
  );

  blockchain.storageItem.delete = jest.fn(async ({ hash, key }) => {
    if (storage[common.uInt160ToString(hash)] !== undefined) {
      delete storage[common.uInt160ToString(hash)][key.toString('hex')];
    }
  });

  blockchain.action.add = jest.fn(async () => Promise.resolve());

  return blockchain;
};
