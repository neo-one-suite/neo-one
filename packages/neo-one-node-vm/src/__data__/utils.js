/* @flow */
import { type StorageItem, utils } from '@neo-one/client-core';
import { utils as commonUtils } from '@neo-one/utils';

export const verifyBlockchainSnapshot = (blockchain: any) => {
  commonUtils.values(blockchain).forEach((obj) => {
    if (typeof obj === 'object') {
      commonUtils.values(obj).forEach((maybeMock) => {
        if (maybeMock != null && maybeMock.mock != null) {
          expect(maybeMock.mock.calls).toMatchSnapshot();
        }
      });
    }
  });
};

export const verifyListeners = (listeners: any) => {
  commonUtils.values(listeners).forEach((func) => {
    expect(func.mock.calls).toMatchSnapshot();
  });
};

export const expectItemBNEquals = (item: StorageItem, value: string) => {
  expect(utils.fromSignedBuffer(item.value).toString(10)).toEqual(value);
};
