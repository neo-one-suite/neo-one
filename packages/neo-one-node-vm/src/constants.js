/* @flow */
import BN from 'bn.js';
import {
  type ExecutionAction,
  type TriggerType,
  type VMListeners,
  type WriteBlockchain,
} from '@neo-one/node-core';
import {
  type Block,
  type OpCode,
  type ScriptContainer,
  type SysCallName,
  type UInt160,
  type UInt160Hex,
  type VMState,
  common,
} from '@neo-one/client-core';
import type { Monitor } from '@neo-one/monitor';

import type { StackItem } from './stackItem';

export const MAX_STACK_SIZE = 2 * 1024;
export const MAX_INVOCATION_STACK_SIZE = 1024;
export const MAX_ARRAY_SIZE = 1024;
export const MAX_ITEM_SIZE = 1024 * 1024;
export const MAX_SCRIPT_LENGTH = 1024 * 1024;
export const MAX_VOTES = 1024;
export const BLOCK_HEIGHT_YEAR = 2000000;
export const MAX_ASSET_NAME_LENGTH = 1024;
const ratio = 100000;
export const FEES = {
  ONE: new BN(1 * ratio),
  TEN: new BN(10 * ratio),
  TWENTY: new BN(20 * ratio),
  ONE_HUNDRED: new BN(100 * ratio),
  TWO_HUNDRED: new BN(200 * ratio),
  FOUR_HUNDRED: new BN(400 * ratio),
  FIVE_HUNDRED: new BN(500 * ratio),
  ONE_THOUSAND: new BN(1000 * ratio),
};
export const FREE_GAS = common.TEN_FIXED8;

export type ExecutionStack = Array<StackItem>;
export type ExecutionInit = {|
  scriptContainer: ScriptContainer,
  triggerType: TriggerType,
  action: ExecutionAction,
  listeners: VMListeners,
  skipWitnessVerify: boolean,
  persistingBlock?: Block,
|};
type CreatedContracts = { [hash: UInt160Hex]: UInt160 };
export type Options = {|
  depth: number,
  stack: ExecutionStack,
  stackAlt: ExecutionStack,
  createdContracts: CreatedContracts,
  scriptHash: ?UInt160,
  entryScriptHash: UInt160,
|};
export type ExecutionContext = {|
  state: VMState,
  errorMessage?: string,
  blockchain: WriteBlockchain,
  init: ExecutionInit,
  engine: {
    run: (input: {| monitor: Monitor, context: ExecutionContext |}) => Promise<
      ExecutionContext,
    >,
    executeScript: (input: {|
      monitor: Monitor,
      code: Buffer,
      pushOnly?: boolean,
      blockchain: WriteBlockchain,
      init: ExecutionInit,
      gasLeft: BN,
      options?: Options,
    |}) => Promise<ExecutionContext>,
  },
  code: Buffer,
  pushOnly: boolean,
  scriptHash: UInt160,
  callingScriptHash: ?UInt160,
  entryScriptHash: UInt160,
  pc: number,
  depth: number,
  stack: ExecutionStack,
  stackAlt: ExecutionStack,
  gasLeft: BN,
  createdContracts: CreatedContracts,
|};

export const getResultContext = (context: ExecutionContext) => ({
  stack: context.stack,
  stackAlt: context.stackAlt,
  gasLeft: context.gasLeft,
  createdContracts: context.createdContracts,
});

export type OpResult = {|
  context: ExecutionContext,
  results?: Array<StackItem>,
  resultsAlt?: Array<StackItem>,
|};
export type OpInvokeArgs = {|
  monitor: Monitor,
  context: ExecutionContext,
  args: Array<StackItem>,
  argsAlt: Array<StackItem>,
|};
export type OpInvoke = (input: OpInvokeArgs) => Promise<OpResult> | OpResult;
export type Op = {|
  name: OpCode,
  in: number,
  inAlt: number,
  out: number,
  outAlt: number,
  modify: number,
  modifyAlt: number,
  invocation: number,
  array: number,
  item: number,
  fee: BN,
  invoke: OpInvoke,
  context: ExecutionContext,
|};

export type SysCall = {|
  name: SysCallName,
  in: number,
  inAlt: number,
  out: number,
  outAlt: number,
  modify: number,
  modifyAlt: number,
  invocation: number,
  array: number,
  item: number,
  fee: BN,
  invoke: OpInvoke,
  context: ExecutionContext,
|};
