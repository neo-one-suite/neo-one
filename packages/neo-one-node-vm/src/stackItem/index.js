/* @flow */
export { default as ArrayStackItem } from './ArrayStackItem';
export { default as BooleanStackItem } from './BooleanStackItem';
export { default as BufferStackItem } from './BufferStackItem';
export {
  default as ConsensusPayloadStackItem,
} from './ConsensusPayloadStackItem';
export { default as UInt160StackItem } from './UInt160StackItem';
export { default as UInt256StackItem } from './UInt256StackItem';
export { default as IntegerStackItem } from './IntegerStackItem';
export { default as HeaderStackItem } from './HeaderStackItem';
export { default as BlockStackItem } from './BlockStackItem';
export { default as ContractStackItem } from './ContractStackItem';
export { default as EnumeratorStackItem } from './EnumeratorStackItem';
export { default as TransactionStackItem } from './TransactionStackItem';
export { default as AccountStackItem } from './AccountStackItem';
export { default as AssetStackItem } from './AssetStackItem';
export { default as AttributeStackItem } from './AttributeStackItem';
export { default as InputStackItem } from './InputStackItem';
export { default as IteratorStackItem } from './IteratorStackItem';
export { default as MapStackItem } from './MapStackItem';
export { default as OutputStackItem } from './OutputStackItem';
export { default as ValidatorStackItem } from './ValidatorStackItem';
export { default as StorageContextStackItem } from './StorageContextStackItem';
export { default as ECPointStackItem } from './ECPointStackItem';
export { default as StructStackItem } from './StructStackItem';
export { default as StackItemEnumerator } from './StackItemEnumerator';
export { default as StackItemIterator } from './StackItemIterator';

export { default as deserializeStackItem } from './deserializeStackItem';

export type { StackItem } from './StackItem';

export * from './StackItemType';
