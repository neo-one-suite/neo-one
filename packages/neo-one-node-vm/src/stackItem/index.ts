export { ArrayStackItem } from './ArrayStackItem';
export { BooleanStackItem } from './BooleanStackItem';
export { BufferStackItem } from './BufferStackItem';
export { ConsensusPayloadStackItem } from './ConsensusPayloadStackItem';
export { UInt160StackItem } from './UInt160StackItem';
export { UInt256StackItem } from './UInt256StackItem';
export { IntegerStackItem } from './IntegerStackItem';
export { HeaderStackItem } from './HeaderStackItem';
export { BlockStackItem } from './BlockStackItem';
export { ContractStackItem } from './ContractStackItem';
export { EnumeratorStackItem } from './EnumeratorStackItem';
export { TransactionStackItem } from './TransactionStackItem';
export { AccountStackItem } from './AccountStackItem';
export { AssetStackItem } from './AssetStackItem';
export { AttributeStackItem } from './AttributeStackItem';
export { InputStackItem } from './InputStackItem';
export { IteratorStackItem } from './IteratorStackItem';
export { MapStackItem } from './MapStackItem';
export { OutputStackItem } from './OutputStackItem';
export { ValidatorStackItem } from './ValidatorStackItem';
export { StorageContextStackItem } from './StorageContextStackItem';
export { ECPointStackItem } from './ECPointStackItem';
export { StructStackItem } from './StructStackItem';
export { StackItemEnumerator } from './StackItemEnumerator';
export { StackItemIterator } from './StackItemIterator';
export { WitnessStackItem } from './WitnessStackItem';
export { NullStackItem } from './NullStackItem';

export { deserializeStackItem } from './deserializeStackItem';
export { StackItem } from './StackItem';

export * from './StackItemType';
