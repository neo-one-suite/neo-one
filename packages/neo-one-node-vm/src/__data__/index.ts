import * as assets from './assets';
import * as transactions from './transactions';
import * as testUtils from './utils';

export * from './createBlockchain';
export * from './keys';

export { assets, testUtils, transactions };
