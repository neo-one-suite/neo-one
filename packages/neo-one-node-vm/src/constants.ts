import { common, OpCode, SysCallName, UInt160, VMState } from '@neo-one/client-common';
import { Block, ExecutionAction, ScriptContainer, TriggerType, VMListeners, WriteBlockchain } from '@neo-one/node-core';
import { BN } from 'bn.js';
import { StackItem } from './stackItem';

export const MAX_SHL_SHR = 65535;
export const MIN_SHL_SHR = -MAX_SHL_SHR;
export const MAX_SIZE_BIG_INTEGER = 32;
export const MAX_STACK_SIZE = 2 * 1024;
export const MAX_ITEM_SIZE = 1024 * 1024;
export const MAX_MANIFEST_LENGTH = 2048;
export const MAX_INVOCATION_STACK_SIZE = 1024;
export const MAX_ARRAY_SIZE = 1024;
export const MAX_ARRAY_SIZE_BN = new BN(1024);
export const BLOCK_HEIGHT_YEAR = 2000000;
export const BLOCK_HEIGHT_MAX_SIZE_CHECKS = Number.MAX_SAFE_INTEGER;
export const MAX_PAYLOAD_SIZE = new BN(0x02000000);
export const GAS_PER_BYTE = new BN(100000);

export const FEES = {
  0: new BN(0),
  30: new BN(30),
  40: new BN(40),
  60: new BN(60),
  70: new BN(70),
  100: new BN(100),
  120: new BN(120),
  150: new BN(150),
  180: new BN(180),
  200: new BN(200),
  250: new BN(250),
  300: new BN(300),
  400: new BN(400),
  500: new BN(500),
  7_000: new BN(7000),
  10_000: new BN(10000),
  13_000: new BN(13000),
  15_000: new BN(15000),
  22_000: new BN(22000),
  30_000: new BN(30000),
  70_000: new BN(70000),
  80_000: new BN(80000),
  100_000: new BN(100000),
  110_000: new BN(110000),
  270_000: new BN(270000),
  300_000: new BN(300000),
  500_000: new BN(500000),
  1_000_000: new BN(1000000),
  2_500_000: new BN(2500000),
  3_000_000: new BN(3000000),
  8_000_000: new BN(8000000),
};

export const FREE_GAS = common.TEN_FIXED8;
export type ExecutionStack = readonly StackItem[];
export interface ExecutionInit {
  readonly scriptContainer: ScriptContainer;
  readonly triggerType: TriggerType;
  readonly action: ExecutionAction;
  readonly listeners: VMListeners;
  readonly skipWitnessVerify: boolean;
  readonly persistingBlock?: Block;
}

export interface CreatedContracts {
  readonly [hash: string]: UInt160;
}
export type InvocationCounter = Record<string, number | undefined>;

export interface VMNotification {
  readonly scriptHash: UInt160;
  readonly args: StackItem;
}

export interface Options {
  readonly depth: number;
  readonly stack: ExecutionStack;
  readonly stackAlt: ExecutionStack;
  readonly createdContracts: CreatedContracts;
  readonly scriptHashStack: readonly UInt160[];
  readonly scriptHash: UInt160 | undefined;
  readonly entryScriptHash: UInt160;
  readonly returnValueCount: number;
  readonly stackCount: number;
  readonly notifications: readonly VMNotification[];
  readonly invocationCounter: InvocationCounter;
  readonly pc?: number;
}
export interface ExecutionContext {
  readonly state: VMState;
  readonly errorMessage?: string;
  readonly blockchain: WriteBlockchain;
  readonly init: ExecutionInit;
  readonly engine: {
    readonly run: (input: { readonly context: ExecutionContext }) => Promise<ExecutionContext>;
    readonly executeScript: (input: {
      readonly code: Buffer;
      readonly blockchain: WriteBlockchain;
      readonly init: ExecutionInit;
      readonly gasLeft: BN;
      readonly options?: Options;
    }) => Promise<ExecutionContext>;
  };
  readonly notifications: readonly VMNotification[];
  readonly code: Buffer;
  readonly scriptHashStack: readonly UInt160[];
  readonly scriptHash: UInt160;
  readonly callingScriptHash: UInt160 | undefined;
  readonly entryScriptHash: UInt160;
  readonly pc: number;
  readonly depth: number;
  readonly stack: ExecutionStack;
  readonly stackAlt: ExecutionStack;
  readonly gasLeft: BN;
  readonly createdContracts: CreatedContracts;
  readonly returnValueCount: number;
  readonly stackCount: number;
  readonly invocationCounter: InvocationCounter;
}

export interface OpResult {
  readonly context: ExecutionContext;
  readonly results?: readonly StackItem[];
  readonly resultsAlt?: readonly StackItem[];
}
export interface OpInvokeArgs {
  readonly context: ExecutionContext;
  readonly args: readonly StackItem[];
  readonly argsAlt: readonly StackItem[];
}
export type OpInvoke = (input: OpInvokeArgs) => Promise<OpResult> | OpResult;
export interface Op {
  readonly name: OpCode;
  readonly in: number;
  readonly inAlt: number;
  readonly out: number;
  readonly outAlt: number;
  readonly invocation: number;
  readonly fee: BN;
  readonly invoke: OpInvoke;
}
export interface SysCall {
  readonly name: SysCallName;
  readonly in: number;
  readonly inAlt: number;
  readonly out: number;
  readonly outAlt: number;
  readonly invocation: number;
  readonly fee: BN;
  readonly invoke: OpInvoke;
  readonly context: ExecutionContext;
}
