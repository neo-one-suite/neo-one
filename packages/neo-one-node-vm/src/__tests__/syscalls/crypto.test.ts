// tslint:disable no-object-mutation
import { crypto } from '@neo-one/client-common';
import { BN } from 'bn.js';
import { keys, runSysCalls, TestCase } from '../../__data__';
import { FEES } from '../../constants';
import { BooleanStackItem } from '../../stackItem';

const signature0 = crypto.sign({
  message: Buffer.alloc(32, 10),
  privateKey: keys[0].privateKey,
});

const signature1 = crypto.sign({
  message: Buffer.alloc(32, 10),
  privateKey: keys[1].privateKey,
});

const SYSCALLS: readonly TestCase[] = [
  {
    name: 'Neo.Crypto.ECDsaVerify',
    result: [new BooleanStackItem(true)],
    args: [keys[0].publicKey, signature0],
    mockTransaction: ({ transaction }) => {
      transaction.messageInternal = jest.fn(() => Buffer.alloc(32, 10));
    },
    gas: FEES[1_000_000],
  },

  {
    name: 'Neo.Crypto.ECDsaVerify',
    result: [new BooleanStackItem(false)],
    args: [keys[0].publicKey, Buffer.alloc(64, 10)],
    gas: FEES[1_000_000],
  },

  {
    name: 'Neo.Crypto.ECDsaCheckMultiSig',
    result: [new BooleanStackItem(true)],
    args: [
      [keys[0].publicKey, keys[1].publicKey],
      [signature0, signature1],
    ],
    mockTransaction: ({ transaction }) => {
      transaction.messageInternal = jest.fn(() => Buffer.alloc(32, 10));
    },
    gas: FEES[1_000_000].mul(new BN(2)),
  },

  {
    name: 'Neo.Crypto.ECDsaCheckMultiSig',
    result: [new BooleanStackItem(true)],
    args: [new BN(2), keys[0].publicKey, keys[1].publicKey, new BN(2), signature0, signature1],
    mockTransaction: ({ transaction }) => {
      transaction.messageInternal = jest.fn(() => Buffer.alloc(32, 10));
    },
    gas: FEES[1_000_000].mul(new BN(2)),
  },

  {
    name: 'Neo.Crypto.ECDsaCheckMultiSig',
    result: [new BooleanStackItem(true)],
    args: [
      [keys[0].publicKey, keys[2].publicKey, keys[1].publicKey],
      [signature0, signature1],
    ],
    mockTransaction: ({ transaction }) => {
      transaction.messageInternal = jest.fn(() => Buffer.alloc(32, 10));
    },
    gas: FEES[1_000_000].mul(new BN(3)),
  },

  {
    name: 'Neo.Crypto.ECDsaCheckMultiSig',
    result: [new BooleanStackItem(true)],
    args: [new BN(3), keys[0].publicKey, keys[2].publicKey, keys[1].publicKey, new BN(2), signature0, signature1],
    mockTransaction: ({ transaction }) => {
      transaction.messageInternal = jest.fn(() => Buffer.alloc(32, 10));
    },
    gas: FEES[1_000_000].mul(new BN(3)),
  },

  {
    name: 'Neo.Crypto.ECDsaCheckMultiSig',
    result: [new BooleanStackItem(false)],
    args: [[keys[0].publicKey, keys[1].publicKey], [Buffer.alloc(64, 10)]],
    gas: FEES[1_000_000].mul(new BN(2)),
  },

  {
    name: 'Neo.Crypto.ECDsaCheckMultiSig',
    result: [new BooleanStackItem(false)],
    args: [new BN(2), keys[0].publicKey, keys[1].publicKey, new BN(1), Buffer.alloc(64, 10)],
    mockTransaction: ({ transaction }) => {
      transaction.messageInternal = jest.fn(() => Buffer.alloc(32, 10));
    },
    gas: FEES[1_000_000].mul(new BN(2)),
  },
];

describe('SysCalls: Neo.Crypto', () => {
  runSysCalls(SYSCALLS);
});
