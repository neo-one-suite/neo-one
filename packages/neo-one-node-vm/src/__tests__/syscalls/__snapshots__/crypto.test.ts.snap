// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 1`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 2`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 3`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 4`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 5`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 6`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 7`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 8`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 9`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 10`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 11`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 12`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 13`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 14`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 15`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 16`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 17`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 18`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 19`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 20`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 21`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 22`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 23`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaCheckMultiSig 24`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 1`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 2`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 3`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 4`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 5`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 6`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 7`] = `Array []`;

exports[`SysCalls: Neo.Crypto Neo.Crypto.ECDsaVerify 8`] = `Array []`;
